Write evaluation section

daffidwilde opened the pull request:
- [x] What is evaluation?
- [x] How are clustering algorithms evaluated?
- [x] How are healthcare models evaluated?

daffidwilde committed:
Update actions to run on PRs and main pushes.

daffidwilde committed:
Rewrite evaluation opening, start on clustering.

daffidwilde committed:
Write majority of clustering section.

daffidwilde committed:
Add interoperable back to known words.

daffidwilde committed:
Finish clustering section.

daffidwilde committed:
Add in mathematical axiomatic form as comments

Not sure if this is overly complicated, but I don't read the definitions
nicely as prose. Then again, are they even necessary?

daffidwilde committed:
Merge branch 'vk-review-2020-12-08' into write-evaluation-section

daffidwilde committed:
Sort the bibliography.

daffidwilde committed:
Start on healthcare section.

daffidwilde committed:
Write TODO list and catch known words.

drvinceknight commented on sec/review.tex:
```diff
@@ -762,36 +765,249 @@ \subsubsection{Queuing and clustering}
+        distances are decreased and between-cluster distances are increased
+        %for every \(C\)-consistent distance metric, \(d' \in \left\{d' \in D_X
+        %| d'(x, y) \le d(x, y) \forall x \sim_C y \wedge d'(x, y) \ge d(x, y)
+        %\forall x \nsim_C y\right\}\), it follows that \(f(d) = f(d')\)
```
```suggestion
       % TODO Replace with mathematical definitions if appropriate.
        %for every \(d \in D_X\) and for all \(\alpha > 0\) it follows that
        %\(f(d) = f(\alpha d)\)
    \item Richness: all partitions of the dataset are possible outputs of the
        function (by use of an alternative distance function)
        % TODO Replace with mathematical definitions if appropriate.
        %for every \(C \in \mathcal B(X)\) there exists some \(d \in \mathcal
        %D_X\) such that \(f(d) = C\)
    \item Consistency: the outputted clustering is unchanged when within-cluster
        distances are decreased and between-cluster distances are increased
        % TODO Replace with mathematical definitions if appropriate.
        %for every \(C\)-consistent distance metric, \(d' \in \left\{d' \in D_X
        %| d'(x, y) \le d(x, y) \forall x \sim_C y \wedge d'(x, y) \ge d(x, y)
        %\forall x \nsim_C y\right\}\), it follows that \(f(d) = f(d')\)
```

Just to document this tweak.

drvinceknight commented on sec/review.tex:
```diff
@@ -762,36 +765,249 @@ \subsubsection{Queuing and clustering}
+definition~\cite{Baumberger2017,Caswell1976,Currie2017}. The author suggests
+that model evaluation should be considered as the process of determining whether
+a model is \emph{adequate} for a purpose, i.e. whether a model is highly likely
+to achieve a goal. The author distinguishes this concept of adequacy from
+individual success where a model achieves a purpose in a particular instance.
```
Paul and I have a paper where we fitted a phase type distribution to a length of stay. I think the "best" fit had 12 phases (or something like that) but when we then put that distribution in to a healthcare queueing model, the decision that was made was the same for a model with just 4 phases. We argued about the importance of not doing more than was necessary.

Not suggesting this is a necessary reference to include here but it could be of interest: http://orca.cf.ac.uk/25345/

daffidwilde replied:
I like this. That's a good example :+1:

daffidwilde replied:
Added in 5aaf115

drvinceknight commented on sec/review.tex:
```diff
@@ -762,36 +765,249 @@ \subsubsection{Queuing and clustering}
+\subsubsection{Clustering}\label{subsubsec:clustering_evaluation}
+
+Algorithms for clustering, like many machine learning tasks, fall into the
+category of models that are primarily evaluated using confirmation processes. To
+reiterate, a confirmation process provides support for a model based on its
```
Instead of `To reiterate` can we say: `As discussed in <section ...>, a confirmation ...`

daffidwilde replied:
Thoughts on this? The section isn't independently numbered

```suggestion
category of models that are primarily evaluated using confirmation processes. As
discussed at the beginning of this section, a confirmation process provides
support for a model based on its
```

drvinceknight replied:
Yeah that's great :+1:

drvinceknight commented on sec/review.tex:
```diff
@@ -762,36 +765,249 @@ \subsubsection{Queuing and clustering}
+such as~\cite{Agrawal1998,Aljarah2019,Bakr2015,Cao2009,Huang1998}.
+
+Information retrieval metrics can be considered \emph{external criteria} in that
+they measure an aspect of a model's clustering according to a external labelling
```
```suggestion
they measure an aspect of a model's clustering according to external labelling
```

daffidwilde replied:
```suggestion
they measure an aspect of a model's clustering according to an external labelling
```

drvinceknight commented on sec/review.tex:
```diff
@@ -762,36 +765,249 @@ \subsubsection{Queuing and clustering}
+
+Information retrieval metrics can be considered \emph{external criteria} in that
+they measure an aspect of a model's clustering according to a external labelling
+of the dataset~\cite{Mitchell1997}. By evaluating a clustering algorithm based
+on a ground truth, the algorithm is effectively being evaluated as a classifier
+--- a method which is trained using that ground truth. Hence, the quality of any
+clustering is predicated on the data generation process, and if a clustering
```
```suggestion
of the dataset~\cite{Mitchell1997}. The quality of any
clustering is predicated on the data generation process, and if a clustering
```

I found that sentence a bit heavy, I wonder if this suggestion is as effective.

daffidwilde replied:
I agree. Will delete.

drvinceknight commented on sec/review.tex:
```diff
@@ -762,36 +765,249 @@ \subsubsection{Queuing and clustering}
+In~\cite{Kleinberg2002}, the author defines clustering as a function, \(f\),
+over a dataset, \(X\), which maps a distance metric, \(d\), also over \(X\) to
+some partition, \(C\), of \(X\).
+%i.e. a clustering function can be defined as
+%any \(f: \mathcal D_X \to \mathcal B(X)\) where \(\mathcal D_X\) is the set of
+%all distance metrics over \(X\) and \(\mathcal B(X)\) is the set of all
+%partitions of \(X\).
```
```suggestion
% TODO Replace with mathematical definitions if appropriate.
%i.e. a clustering function can be defined as
%any \(f: \mathcal D_X \to \mathcal B(X)\) where \(\mathcal D_X\) is the set of
%all distance metrics over \(X\) and \(\mathcal B(X)\) is the set of all
%partitions of \(X\).
```

drvinceknight commented on sec/review.tex:
```diff
@@ -762,36 +765,249 @@ \subsubsection{Queuing and clustering}
+considered as a generalisation of the silhouette coefficient with respect to its
+loss function or distance measure.
+
+% TODO do we need a brief discussion of datasets? Or should we leave that for
```
Are you thinking about biases in training data?

daffidwilde replied:
Not exclusively. I'm not entirely sure what I'd want to discuss here (thinking for the chapter) as I go into the issues with benchmarks and dataset reliability in the EDO chapter.

For the paper, I think there should be *something* about datasets here.

drvinceknight replied:
Sure, I'm not entirely sure what you'd have here but in healthcare a common stated "problem" is lack of data for validation of models. That could perhaps go here (as well as bias in data)

daffidwilde replied:
That is something I am going to touch on the last section (healthcare evaluation). Something along the lines of:

- an important point to raise about evaluation of healthcare models are their longterm effects in terms of trust
- point at some of the ethics papers specifically about trustworthiness, and the growing call for unity in best practices for healthcare models (academic and clinical)
- as part of that, one of the key stakeholders in healthcare is the patient (link back to patient-centred care) and so building trust for patients is essential for adopting data-driven decision making in modern healthcare systems
- a crucial component of the patient-system trust relationship regards patient privacy (I have a reference about GDPR in healthcare, and I think there's an article somewhere about Google DeepMind accidentally leaking health data)
- to circumvent these issues, pseudo-privacy can be obtained through data synthesis (as well as anonymisation)
- add on lack of data for assessment, "two birds one stone"-style

drvinceknight replied:
That sounds good, those bullet points with a bit of flesh here and there are probably exactly what you need :+1:

daffidwilde committed:
Add in TODOs for comments

Some alternative statements have been left as comments; this adds in a preceding `TODO` so they aren't missed.

Co-authored-by: Vince Knight <vincent.knight@gmail.com>

daffidwilde committed:
Apply VK editing suggestions

Co-authored-by: Vince Knight <vincent.knight@gmail.com>

drvinceknight committed:
Vk review 2020 12 08 (#6)

* Discuss multi-class queues + intersection.

* Finish intersection of queuing and clustering.

* Make some comments/suggestions on intro.tex

* Make some comments/suggestions on review section.

* Implement smaller changes to introduction.

* Implement minor changes to review.

Mainly citation fixes, rewording and adding in pointers.

* Add `amsmath` to packages.

* Fix Bilu2012 citation.

* Add URL for DoH2011 citation.

* Add references for software best practices.

* Add in missing Steins2013 citation.

* Checkout workflow file from main.

* Checkout workflow file from main.

* Add citations for healthcare modelling reviews.

* Add textbook reference for elbow method paragraph.

* Add signpost for clustering in healthcare.

* Add scikit-learn to environment.

* Write notebook for plots.

* Create image files.

* Enhance the plots

- Add Voronoi cells for kmeans
- Remove axis for dendograms

* Add in plots as figures.

* Add dataset summary table.

* Make scatter marker larger.

* Apply notebook formatters.

* Add descriptions of each figure; add true labels.

The `true' labels for the clusters have been added to the scatter plots
as different markers. This more readily shows any mis-clusterings.

* Move to k-means-sharp.

* Remove citations from start of sentences.

* Fix reference to healthcare section.

* Reword sentences leading with citations.

Only in the closing paragraphs of the queuing section.

* Add analytics to known words.

* Fix titles in bibliography.

* Fix linebreaks in text.

* Fix typo in known words.

* Add structure for archiving.

* Ignore csv files.

* Write data generation source code.

* Add the plot source code.

* Update plots with new source code.

* Update figure file locations.

* Ignore zip archives.

* Add TODOs for GH and DOI.

Co-authored-by: Vince Knight <vincent.knight@gmail.com>

* Add pointers and explanation of label omission.

* Remove edgelines from kmeans moons plot.

* Update cluster image DOI.

* /common sense resort/common resort/

Co-authored-by: Vince Knight <vincent.knight@gmail.com>

* Add in requested citations.

* Add quantise to known words.

Co-authored-by: Henry Wilde <henrydavidwilde@gmail.com>

daffidwilde committed:
Rephrase confirmation process pointer

daffidwilde committed:
Merge branch 'main' into write-evaluation-section

daffidwilde commented:
I've opened a [new PR on my fork](https://github.com/daffidwilde/literature-review/pull/8/) now that the PR for `vk-review-2020-12-08` has been merged. Given that there are only two points left here, let's continue discussion for that new PR here.

drvinceknight commented:
> I've opened a [new PR on my fork](https://github.com/daffidwilde/literature-review/pull/8/) now that the PR for `vk-review-2020-12-08` has been merged. Given that there are only two points left here, let's continue discussion for that new PR here.

I don't fully follow what's happening with forks (I don't fully understand why this is happening on my fork at all but I'm content ) any more. So I'm just going to respond wherever you message me, but ping me if that's not what you want.

daffidwilde committed:
Flesh out healthcare evaluation section.

daffidwilde committed:
Merge branch 'vk-review-2020-12-08' into write-evaluation-section

daffidwilde committed:
Add missing Horn2015 citation.

daffidwilde committed:
Clean up bibtex warnings in bibliography.

daffidwilde committed:
Finish healthcare evaluation section.

daffidwilde committed:
Add missing words and extra citations.

daffidwilde committed:
Add adequacy example paragraph.

drvinceknight commented on sec/review.tex:
```diff
@@ -1081,8 +1078,19 @@ \subsubsection{Healthcare settings}
+models~\cite{Dahmen2019,Dube2014,McLachlan2016,Tucker2020} and, recently,
+generative adversarial networks (GANs)~\cite{Avino2018,Torfi2020}. Using
+synthetic datasets provides another level of anonymity to the patients, but
+their efficacy against real data has been contested. In a recent
```
I'm not sure what `efficacy against real data` means here. Would something like: 

```suggestion
their but their validity as representations of real data has been contested. In a recent
```

work?

daffidwilde replied:
That's much clearer, yes.
